client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -118,6 +118,46 @@
      */
     CompletableFuture<Boolean> updateStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
 
+    /**
+     * API to add a Subscriber to the Stream.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @param subscriber Name/Id that uniquely identifies a Stream Subscriber.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the subscriber was updated in Stream Metadata.
+     */
+    CompletableFuture<Boolean> addSubscriber(final String scope, final String streamName, final String subscriber);
+
+    /**
+     * API to remove a Subscriber from list of Subscribers for the Stream.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @param subscriber Name/Id that uniquely identifies a Stream Subscriber..
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the subscriber was updated in Stream Metadata.
+     */
+    CompletableFuture<Boolean> removeSubscriber(final String scope, final String streamName, final String subscriber);
```

Comment:
same question: what is the reason to return a boolean here?

Reply:
This is a client interface and every operation on Controller is expected to fail or succeed.
There is no _return status_ for other operations as well.
Only the getXXX operations return non-boolean values.